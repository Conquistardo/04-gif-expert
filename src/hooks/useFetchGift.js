import { useState, useEffect } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGift = (categoria) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifs(categoria)
        .then(imgs => {
            setTimeout( () => {
                setState({
                    data: imgs,
                    loading: false
                })
            },
            2000);
        });
    }, [categoria])

    return state;
}
