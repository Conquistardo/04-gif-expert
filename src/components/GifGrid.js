import React from 'react'
import { useFetchGift } from '../hooks/useFetchGift';
import { GifGridItems } from './GifGridItems';
import PropTypes from 'prop-types';

export const GifGrid = ({categoria}) => {

    const {data:images, loading} = useFetchGift(categoria);

    return (
        <>
            <h3>{categoria}</h3>
            {loading?<p>Cargando...</p>:''}
            <div className='cardGrid'>            
                    {
                        images.map(
                            (img) => <GifGridItems 
                                    key={img.id}
                                    {...img}/>
                        )
                    }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    categoria: PropTypes.string.isRequired
}
