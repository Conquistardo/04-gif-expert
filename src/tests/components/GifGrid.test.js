import { GifGrid } from "../../components/GifGrid";
import {shallow} from 'enzyme';
import { useFetchGift } from "../../hooks/useFetchGift";
jest.mock('../../hooks/useFetchGift');

describe('Pruebas en el componente GifGrid', () => {
    const categoria = 'Ultra instinto';
    
    test('debe de hacer match en snapshot', () => {
        useFetchGift.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid categoria={categoria}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifts', () => {
        const imgs = [{
            id: 'test123',
            url: 'testurl',
            title: 'testtitle'
        }]
        useFetchGift.mockReturnValue({
            data: imgs,
            loading: false
        });
        const wrapper = shallow(<GifGrid categoria={categoria}/>);

        //expect(wrapper.find('p').exists).toBe(false);//Est√° prueba no puede pasar ya que el tag 'p' si existe en coponente GifGridItems
        expect(wrapper.find('GifGridItems').length).toBe(imgs.length);
    });
    
    
});