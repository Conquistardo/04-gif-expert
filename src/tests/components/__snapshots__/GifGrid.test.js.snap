// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid debe de hacer match en snapshot 1`] = `
<Fragment>
  <h3>
    Ultra instinto
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="cardGrid"
  />
</Fragment>
`;
